diff --git a/cmake/OptimalOptions.cmake b/cmake/OptimalOptions.cmake
index eba8624..f2714e4 100644
--- a/cmake/OptimalOptions.cmake
+++ b/cmake/OptimalOptions.cmake
@@ -1,7 +1,9 @@
 
 # Set optimal options for gcc:
 if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-	SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=native")
+	if (NOT CMAKE_SYSTEM_NAME STREQUAL "iOS")
+		SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=native")
+	endif()
 	SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
 ENDIF()
 
